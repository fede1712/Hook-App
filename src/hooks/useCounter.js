import { useState } from "react"

export const useCounter = (initialState = 0) => {

    const [counter, setCounter] = useState(initialState)

    const add = () => {
        setCounter(counter + 1)
    }

    const substract = () => {
        setCounter(counter - 1)
    }

    const reset = () => {
        setCounter(initialState)
    }

    return {
        counter,
        add,
        substract,
        reset
    }
}
