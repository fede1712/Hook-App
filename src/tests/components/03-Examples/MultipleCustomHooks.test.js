import React from 'react'
import { shallow } from 'enzyme'

import { MultipleCustomHooks } from "../../../components/03-Examples/MultipleCustomHooks"
import { useFetch } from '../../../hooks/useFetch'
jest.mock('../../../hooks/useFetch')

describe('Tests in <MultipleCustomHooks />', () => {



    test('should render <MultipleCustomHooks /> correctly', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />)

        expect(wrapper).toMatchSnapshot()

    })

    test('should show the info', () => {

        useFetch.mockReturnValue({
            data: [{ author: 'Fede', quote: 'Hola mundo' }],
            loading: false,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks />)

        expect(wrapper.find('.alert').exists()).toBe(false)
        expect(wrapper.find('.mb-0').text().trim()).toBe('Hola mundo')
        expect(wrapper.find('footer').text().trim()).toBe('Fede')
    })



})
