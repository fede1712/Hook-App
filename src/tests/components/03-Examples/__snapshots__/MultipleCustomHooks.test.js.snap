// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <MultipleCustomHooks /> should render <MultipleCustomHooks /> correctly 1`] = `
<Fragment>
  <h1>
    Breaking Bad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text center"
  >
    Loading...
  </div>
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Next Qoute
  </button>
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Previous Qoute
  </button>
</Fragment>
`;
